src/devtools/views/Components/TreeContext.js

```diff
@@ -298,10 +299,25 @@ function reduceSearchState(store: Store, state: State, action: Action): State {
             if (prevSearchIndex === null) {
               searchIndex = 0;
             } else {
-              searchIndex = Math.min(
-                ((prevSearchIndex: any): number),
-                searchResults.length - 1
-              );
+              // Changes in search index or typing should override the selected element.
+              // The one exception is when a search is broadened (e.g. "Cat" -> "Ca").
```

Comment:
You might be editing anywhere in a search query, not just the end. To me, it would seem a slightly better heuristic would be:

When the search query changes…
1. If the currently selected element is one of the search results, keep the selection unchanged.
2. Otherwise, move the selection to the first result.

Reply:
Ah good point. I haven’t thought about this when I added it.

Reply:
Fixing this in https://github.com/bvaughn/react-devtools-experimental/pull/101.

Reply:
#101 should stack on top of this PR, right?

Reply:
Not sure what you mean. I sent #101 a while ago — it's an independent fix. I just added a new commit to it that helps this particular heuristic. I guess you could do it in either order but I think merging #101 doesn't need to wait for this PR.

Reply:
My bad Dan. I was confused. I overlooked #101 somehow and thought it was a PR you created overnight. 😄  I just looked at it and it looks good, so I'll rebase this on top of it.